Fixed unhelpful error message in from_thread_run functions. (#1513)
python-trio/trio: newsfragments/1244.bugfix.rst

> Added helpful error message in the case the user passed incorrect type of function to "from_thread_run" & "from_thread_run_sync" functions.

Reviewer: ```suggestion
Added a helpful error message if a sync function is passed to `trio.from_thread.run_sync` or an async function to `trio.from_thread.run`.
```

Reworded for clarity and to link to the documentation of those functions (that's what the single backticks do).
